src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotSegment.cs

```diff
@@ -232,5 +234,170 @@ public void WriteToStringBuilder(StringBuilder stringBuilder)
             botAction.WriteToStringBuilder(stringBuilder);
             stringBuilder.Append("}");
         }
+        
+        /**
+         * <summary>
+         * Loads all the Segments that exist in this project (for use in the Editor or in a build)
+         * </summary>
+         */
+        public static List<BotSequenceEntry> LoadAllSegments()
+        {
+            var segments = new List<BotSequenceEntry>();
+            
+#if UNITY_EDITOR
+            const string sequencePath = "Assets/RegressionGames/Resources/BotSegments";
+            if (!Directory.Exists(sequencePath))
+            {
+                return new List<BotSequenceEntry>();
+            }
+
+            segments = LoadSegmentsInDirectory(sequencePath);
+#else
+            // 1. check the persistentDataPath for segments
+            var persistentDataPath = Application.persistentDataPath + "/BotSegments";
+            if (Directory.Exists(persistentDataPath))
+            {
+                segments = LoadSegmentsInDirectoryForRuntime(persistentDataPath);
+            }
+        
+            // 2. load Segments from Resources, and ensure that no Segments have duplicate names
+            const string runtimePath = "BotSegments";
+            segments = segments.Concat(LoadSegmentsInDirectoryForRuntime(runtimePath)).ToList();
+#endif
+
+            return segments;
```

Comment:
We definitely need to make sure to document this well, I was confused last week when trying out this new overlay because I couldn't go and look for my sequences like with the file explorer.

Reply:
Do we want to have both the new UI and file explorer for choosing segments?

Reply:
As per our cycle planning meeting: we're just going to leave the file explorer button where it is for now (on the recording toolbar, outside of the new UI)